songs_app_backend/users/models.py

```diff
@@ -24,3 +30,28 @@ class User(AbstractUser):
 
     def __str__(self):
         return f"{self.username}({self.id})"
+
+    @receiver(reset_password_token_created)
+    def password_reset_token_created(sender, instance, reset_password_token, *args, **kwargs):
+        forgot_password_token = "{}".format(reset_password_token.key)
+        greetings = "Hi {}!".format(reset_password_token.user.username)
+
+        email_html_content = "<html><body><p>{greetings}</p><p>Please use this Token for password Reset on Songs App:<b> {token}</b></p> <p> The token will expires after 60 minutes.</p><p>If you did not ask to reset your password, please ignore this message.</p></body></html>".format(
+            greetings=greetings, token=forgot_password_token)
+
+        message = Mail(
+            from_email=os.getenv('EMAIL_SENDER'),
+            to_emails=[reset_password_token.user.email],
+            subject="Password Reset for {title}".format(title="Songs App. ") +
+            str(datetime.now()),
+            html_content=email_html_content
+        )
+        sendgrid_client = SendGridAPIClient(
+            api_key=os.getenv('SENDGRID_API_KEY'))
+        response = sendgrid_client.send(message)
+
+        if response.status_code == 202:
+            print('Password Reset Email sent successfully to ' +
+                  str(reset_password_token.user.email))
+        else:
+            print('Failed to send email')
```

Comment:
This is right for now but next time try creating another function fir sending the email and other logic and call it here so that the models don't get this messy

Reply:
Done!